pydra/engine/helpers_file.py

```diff
@@ -77,14 +79,26 @@ def hash_file(afile, chunk_len=8192, crypto=sha256, raise_notfound=True):
             raise RuntimeError('File "%s" not found.' % afile)
         return None
 
+    # if the path exists already in precalculated
+    # the time of the last modification will be compared
+    # and the precalculated hash value will be used if the file has not change
+    if precalculated and str(Path(afile)) in precalculated:
+        pre_mtime, pre_cont_hash = precalculated[str(Path(afile))]
+        if Path(afile).stat().st_mtime == pre_mtime:
+            return pre_cont_hash
+
     crypto_obj = crypto()
     with open(afile, "rb") as fp:
         while True:
             data = fp.read(chunk_len)
             if not data:
                 break
             crypto_obj.update(data)
-    return crypto_obj.hexdigest()
+
+    cont_hash = crypto_obj.hexdigest()
+    if precalculated is not None:
+        precalculated[str(Path(afile))] = (Path(afile).stat().st_mtime, cont_hash)
```

Comment:
does this key get deleted somewhere when the path is removed? or `precalculated` is reset?

Reply:
I've decided to reset in `BaseSpec` for each name of the input name, see https://github.com/nipype/pydra/pull/382/files#diff-e76a5941f8f208ec57e47ec8c8233030ced110ae66222044ae5287f52bf65ed2R93

Reply:
perhaps the test could also help when reading the pr: https://github.com/nipype/pydra/pull/382/files#diff-65ddfb82f62778d7637f5040b6d46040a9ee599cd5e42fa78d00dfc333413851R245